src/visualizers/VisualizerCycles.ts

```diff
@@ -0,0 +1,270 @@
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+import {VisualizerDefault} from './VisualizerDefault'
+import {floorSqrt} from '@/shared/math'
+
+/** md
+# cycles
+This visualizer displays the number of steps each sequence value 
+takes to reach a repeating
+cycle under either the Collatz function or the Juggler function.
+**/
+
+//Choices of sequence formula
+enum cycleFormula {
+    Collatz,
+    Juggler,
+}
+
+//Find the number of steps 'n' takes to reach 1 under the collatz formula
+function collatzStepsTo1(n: bigint): number {
+    let steps = 0n
+    if (n <= 1n) {
+        return 1
+    }
+    while (n !== 1n) {
```

Comment:
In this and in juggler, would it safer to check for ```n !<= 1n``` (if that's proper syntax?)?  I know that theoretically the sequences should stay positive, once they are positive, so perhaps this is not needed.  But it might help stop some infinite loop in a buggy situation somewhere.  And it might encourage someone adding another iterator to this someday to program it safely?  I'm curious what @gwhitney and @liammulh would say.

Reply:
Sure, changing the condition to `while (n > 1n)`  (there is no `!<=` but that is logically equivalent to `>` so we're OK ;-) would be a nice, conservative change that could capture weirdness like some part of the computation overflowing if it goes outside of bigint or something weird like that. I don't _think_ it can happen, but there's no reason not to use the more expansive stopping criterion, it's certainly no less safe or less clear, and it might help keep things from going sideways in some weird case.